File: 0009-palindrome-number/0009-palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        
        l = 0
        r = len(str(x))-1
        
        while l < r:
            
            if str(x)[l] != str(x)[r]:
                return False
            
            l += 1
            r -= 1
        
        return True
        

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        
        String x_string = Integer.toString(x);
        int l = 0;
        int r = x_string.length()  - 1;
        
        while (l < r) {
            if (x_string.charAt(l) != x_string.charAt(r)) {
                return false;
            }
            l++;
            r--;
        }
        
        return true;
            
            
        
    }
}